shakenfist/config.py

```diff
@@ -14,142 +13,100 @@
-                            '%(vername)s-server-cloudimg-amd64.img'),
-
-    # NODE SPECIFIC
-    # -------------
-
-    # The bus to use for disk devices. One of virtio, scsi, usb, ide, etc.
-    # See libvirt docs for full list of options.
-    'DISK_BUS': 'virtio',
-
-    # The format for disks. Options are qcow (COW layers onto of image cache),
-    # qcow_flat (just qcow2, no COW), and flat (just raw disk).
-    'DISK_FORMAT': 'qcow',
-
-    # The IP of this node, populated by parse()
-    'NODE_IP': None,
-    'NODE_NAME': None,
-    'NODE_EGRESS_NIC': 'eth0',
-
-    # Where on disk instances are stored
-    'STORAGE_PATH': '/srv/shakenfist',
-
-    # LOGGING
-    # -------
-    'SLOW_LOCK_THRESHOLD': 5.0,
-    'LOGLEVEL_API': 'info',
-    'LOGLEVEL_CLEANER': 'info',
-    'LOGLEVEL_MAIN': 'info',
-    'LOGLEVEL_NET': 'info',
-    'LOGLEVEL_RESOURCES': 'info',
-    'LOGLEVEL_TRIGGERS': 'info',
-    # Add method name and module line number to log messages
-    'LOG_METHOD_TRACE': 0,
-}
-
-
-class Config(object):
-    def __init__(self):
-        self.config = None
-
-    def parse(self):
-        global CONFIG_DEFAULTS
-
-        node_name = socket.getfqdn()
-        try:
-            node_ip = socket.gethostbyname(node_name)
-        except Exception as e:
-            # Only for localhost development environments
-            node_ip = '127.0.0.1'
-            util.ignore_exception('config parser', e)
-
-        CONFIG_DEFAULTS['NODE_NAME'] = node_name
-        CONFIG_DEFAULTS['NODE_IP'] = node_ip
-        CONFIG_DEFAULTS['NETWORK_NODE_IP'] = node_ip
-
-        self.config = copy.copy(CONFIG_DEFAULTS)
-        for var in os.environ:
-            if var.startswith('SHAKENFIST_'):
-                flag = var.replace('SHAKENFIST_', '')
-                value = os.environ[var]
-
-                # We use the type of the default value to determine
-                # what type we should force the value provided by an
-                # environment variable into.
-                if isinstance(CONFIG_DEFAULTS[flag], int):
-                    value = int(value)
-                elif isinstance(CONFIG_DEFAULTS[flag], float):
-                    value = float(value)
-                elif isinstance(CONFIG_DEFAULTS[flag], bool):
-                    value = strtobool(value)
-                elif not isinstance(CONFIG_DEFAULTS[flag], str):
-                    raise exceptions.FlagException(
-                        'Flag %s has unknown type.' % flag)
-
-                self.config[flag] = value
-
-    def get(self, var):
-        if not self.config:
-            self.parse()
-        return self.config.get(var)
-
-
-parsed = Config()
+def get_node_ip():
+    node_name = socket.getfqdn()
+    try:
+        return socket.gethostbyname(node_name)
+    except Exception as e:
+        # Only for localhost development environments
+        util.ignore_exception('config parser', e)
+        return '127.0.0.1'
+
+
+class SfConfig(BaseSettings):
```

Comment:
Am I weird for thinking this class should be called SFConfig? The lower case 'f' jars me for some reason.

Reply:
Changed